brave/brave-core
# Sync remove other device

Resolves https://github.com/brave/brave-browser/issues/11232

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
I. Sync UI is changed on Android
1. Install browser on Android
2. Ensure UI is as shown at https://github.com/brave/brave-core/pull/6732#issuecomment-714525359


II. detection of device which does not support self-delete
1. On put into the same sync chain:
  - device A, desktop, which can do self-delete
  - device B, desktop, which cannot do self-delete
  - device C, Android, which cannot do self-delete    
2. Ensure on device A in the devices list there is no trashcan icon near device B and device C


III. All devices support self-delete
1. Put into the same sync chain, desktop devices A and B, Android device C, which all support self-delete;
2. Ensure on each device in the devices list other devices are displayed with trashcan icon;
3. From device A remove devices B and C from the chain by pressing the trashcan icon near them;
4. Ensure devices B and C are no longer in devices list on the device A;
5. Ensure on devices B and C reset actually happened, sync settings page should want to create or connect to the sync chain;
6. Connect again devices B and C to the same sync chain;
7. From Android device C remove device B from the sync chain;
8. Ensure device B is no longer in devices list on the device A and C;
8. Ensure device B actually did self-reset;


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
